### PR TITLE
Fix kubelet bootstrapping on systemd to not manipulate cgroups

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Following changes:
- The `ContainerManager` bootstrapping is split out to support systemd better.
  - detects the cgroup that the kubelet was launched in
  - defaults 'system.slice' as the system container
  - no longer moves pids around into containers since systemd has everything in a container already
- The non-systemd code is cleaned up to centralize the code that moved the kubelet into a container.

/cc @jonboulle @vishh @dchen1107 @kubernetes/goog-node  @kubernetes/rh-cluster-infra

### CODE REVIEW

**Comment by vishh** [comment]
@derekwaynecarr: I cannot review this PR until 2/8. FYI: For v1.2 release I opened #20687 to make kubelet compatible with systemd.

**Comment by derekwaynecarr** [comment]
@vish - will review referenced PR

File: pkg/kubelet/cm/bootstrap/cgroupfs/bootstrap.go [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+		allPids, err := rootContainer.GetPids()
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to list PIDs for root: %v", err))
+			continue
+		}
+
+		// Remove kernel pids and other protected PIDs (pid 1, PIDs already in system & kubelet containers)
+		pids := make([]int, 0, len(allPids))
+		for _, pid := range allPids {
+			if isKernelPid(pid) {
```
Comment (jimmidyson) [inline pkg/kubelet/cm/bootstrap/cgroupfs/bootstrap.go]: Needs to check if pid 1 as well - shouldn't be moved.

Comment (derekwaynecarr) [inline pkg/kubelet/cm/bootstrap/cgroupfs/bootstrap.go]: cool, just saw this merged that had that change, will update: https://github.com/kubernetes/kubernetes/pull/20395

**Comment by derekwaynecarr** [comment]
@dchen1107 @vishh - I have been getting a number of error reports from users that are getting confused about messages like the following in their logs and I would like to have a solution in place for them that works over the long run.

```
W0209 17:35:58.806107   20834 container_manager_linux.go:272] [ContainerManager] Failed to ensure state of "/docker-daemon": [failed to move PID 20848 (in "/") to "/docker-daemon", failed to move PID 20499 (in "/system.slice/docker.service") to "/docker-daemon"]
W0209 17:36:59.138225   20834 container_manager_linux.go:272] [ContainerManager] Failed to ensure state of "/docker-daemon": [failed to move PID 20848 (in "/") to "/docker-daemon", failed to move PID 20499 (in "/system.slice/docker.service") to "/docker-daemon"]
W0209 17:37:59.313136   20834 container_manager_linux.go:272] [ContainerManager] Failed to ensure state of "/docker-daemon": failed to move PID 20499 (in "/system.slice/docker.service") to "/docker-daemon"
```

I know we are wrestling between various solutions in this PR and in https://github.com/kubernetes/kubernetes/pull/20687 but a solution for 1.2 would be good for everyone.

**Comment by derekwaynecarr** [comment]
Closing this PR since @vishh and I arrived at a solution in https://github.com/kubernetes/kubernetes/pull/20687

**Pull request closed without merging**